File: agentcore/src/main/java/agentcore/agentfoundation/IAgentDatabase.java
package agentcore.agentfoundation;

import agentcore.database.dto.LocalDataDto;
import agentcore.inputdata.ATableDesc;

import javax.annotation.Nonnull;
import java.sql.SQLException;

/**
 * Created by user on 21.02.2017.
 */
public interface IAgentDatabase {

    void addSolution(@Nonnull LocalDataDto dtoEntity) throws SQLException;
    void updateSolution(@Nonnull LocalDataDto dtoEntity) throws SQLException;
    void clearDatabase() throws SQLException;
    @Nonnull ATableDesc getLocalDbTableDesc();

}

File: agentcore/src/main/java/agentcore/agentfoundation/IAgentDatabase.kt
package agentcore.agentfoundation

import agentcore.database.dto.LocalDataDto
import agentcore.inputdata.ATableDesc
import java.sql.SQLException

/**
 * Created by user on 21.02.2017.
 */
interface IAgentDatabase {

    @Throws(SQLException::class)
    fun addSolution(dtoEntity: LocalDataDto)

    @Throws(SQLException::class)
    fun updateSolution(dtoEntity: LocalDataDto)

    @Throws(SQLException::class)
    fun clearDatabase()

    val localDbTableDesc: ATableDesc
}
